src/codeu/chat/server/Model.java

```diff
@@ -67,28 +87,44 @@ public int compare(Time a, Time b) {
   private final Store<Uuid, User> userById = new Store<>(UUID_COMPARE);
   private final Store<Time, User> userByTime = new Store<>(TIME_COMPARE);
   private final Store<String, User> userByText = new Store<>(STRING_COMPARE);
+  private List<User> users = new ArrayList<User>();
```

Comment:
Can these lists be final? A final container can still be modified, it just can't be reassigned.